src/components/tab/tab.test.tsx

```diff
@@ -99,4 +107,28 @@ describe('Tab', () => {
       );
     });
   });
+
+  describe('With Badge', () => {
+    it('should render badge', async () => {
+      sut({ ...defaultTab, badge: { label: 2 } });
+      expect(screen.getAllByTestId('ion-badge'));
+    });
+
+    it('should not render badge when is not informed', async () => {
+      sut();
+      expect(screen.queryAllByText('ion-badge')).toHaveLength(0);
+    });
+
+    it('should render badge with string label', async () => {
+      const label = 'Skywalker';
+      sut({ ...defaultTab, badge: { label: label } });
+      expect(screen.getByTestId('ion-badge').textContent).toBe(label);
```

Comment:
```suggestion
      expect(screen.getByTestId('ion-badge').textContent).toBe(label);
```

You can create a function like getTab.